internal/characters/thoma/burst.go

```diff
@@ -32,90 +36,115 @@ func (c *char) Burst(p map[string]int) action.ActionInfo {
 				Write("frame", c.Core.F)
 			return false
 		}
-		c.AddStatus(icdKey, icd, true)
+		c.summonFieryCollapse()
+		c.Core.Log.NewEvent("thoma burst on state change", glog.LogCharacterEvent, c.Index).
+			Write("frame", c.Core.F).
+			Write("char", c.Core.Player.Active()).
+			Write("icd", c.StatusExpiry(burstICDKey))
+		c.burstTickSrc = c.Core.F
+		c.QueueCharTask(c.burstTickFunc(c.Core.F), 60)
+		return false
+	}, "thoma-burst-animation-check")
+}
 
-		ai := combat.AttackInfo{
-			ActorIndex: c.Index,
-			Abil:       "Fiery Collapse",
-			AttackTag:  combat.AttackTagElementalBurst,
-			ICDTag:     combat.ICDTagElementalBurst,
-			ICDGroup:   combat.ICDGroupDefault,
-			StrikeType: combat.StrikeTypeDefault,
-			Element:    attributes.Pyro,
-			Durability: 25,
-			Mult:       burstproc[c.TalentLvlBurst()],
-			FlatDmg:    0.022 * c.MaxHP(),
+func (c *char) burstTickFunc(src int) func() {
+	return func() {
+		if !c.StatusIsActive(burstKey) {
+			return
 		}
-		//trigger a chain of attacks starting at the first target
-		atk := combat.AttackEvent{
-			Info: ai,
+		if c.burstTickSrc != src {
+			c.Core.Log.NewEvent("thoma burst tick stopped, src diff", glog.LogCharacterEvent, c.Index).
+				Write("src", src).
+				Write("new src", c.burstTickSrc)
+			return
 		}
-		atk.SourceFrame = c.Core.F
-		atk.Pattern = combat.NewDefSingleTarget(t.Index(), combat.TargettableEnemy)
-		cb := func(_ combat.AttackCB) {
-			shieldamt := (burstshieldpp[c.TalentLvlBurst()]*c.MaxHP() + burstshieldflat[c.TalentLvlBurst()])
-			c.genShield("Thoma Burst", shieldamt)
+		state := c.Core.Player.CurrentState()
+		if state != action.NormalAttackState {
+			c.Core.Log.NewEvent("thoma burst tick stopped, not normal state", glog.LogCharacterEvent, c.Index).
+				Write("src", src).
+				Write("state", state)
+			return
 		}
-		atk.Callbacks = append(atk.Callbacks, cb)
-		c.Core.QueueAttackEvent(&atk, 0)
-
-		c.Core.Log.NewEvent("thoma Q proc'd", glog.LogCharacterEvent, c.Index).
-			Write("frame", c.Core.F).
-			Write("char", ae.Info.ActorIndex).
-			Write("attack tag", ae.Info.AttackTag)
+		c.Core.Log.NewEvent("thoma burst triggered from tick", glog.LogCharacterEvent, c.Index).
+			Write("src", src).
+			Write("state", state).
+			Write("icd", c.StatusExpiry(burstICDKey))
+		c.summonFieryCollapse()
+		c.QueueCharTask(c.burstTickFunc(src), 60)
+	}
+}
 
-		return false
-	}, "thoma-burst")
+func (c *char) summonFieryCollapse() {
+	ai := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Fiery Collapse",
+		AttackTag:  combat.AttackTagElementalBurst,
+		ICDTag:     combat.ICDTagElementalBurst,
+		ICDGroup:   combat.ICDGroupDefault,
+		StrikeType: combat.StrikeTypeDefault,
+		Element:    attributes.Pyro,
+		Durability: 25,
+		Mult:       burstproc[c.TalentLvlBurst()],
+		FlatDmg:    0.022 * c.MaxHP(),
+	}
+	// trigger a chain of attacks starting at the first target
+	shieldCb := func(_ combat.AttackCB) {
```

Comment:
not sure if this is intended or not.. but `shieldCb` is called more than once in multi target situations.